test/functional/feature_txindex_compatibility.py

```diff
@@ -25,6 +25,8 @@ def set_test_params(self):
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_previous_releases()
+        # Not all release binaries work with valgrind.
```

Comment:
Thought about that, but there may other (future) tests we want to skip under valgrind (e.g. things that use a brittle dependency). Conversely, since some of tests _do_ work, maybe someone want to enable one later.

Reply:
>  (future) tests we want to skip under valgrind

Let's not worry about unwritten tests, that we apparently might not want to test properly for some arbitrary reason.

Reply:
I was also thinking existing tests, e.g. I haven't been able to run valgrind with BDB configured (didn't try very hard).

Reply:
> Which raises the question how many settings we want --valgrind to be supported under.

I thnk currently the only realistic place we can fully support Valgrind, is within the CI system, where things like BDB are working, and we can more easily control settings and suppressions. Outside of that, things may work fine, or may not at all, depending on the system where it is being run.